### ISSUE TITLE
Cam center in export SVG

### ISSUE THREAD

**Issue by makeCNCWorkshops** [jumpjack/CamFollowerJS#1]
Hello i love this software it will be perfect for making cams for my automata however i have a problem...when i export the Cam as an SVG
there is no reference point or hole where the cam center is and it is  impossible to find it again accurately after export.
is there a way you can add a simple hole for the shaft position of the exported SVG as a reference for where the shaft will go in the cam?
it doesn't matter what size it is as i can use it in my cad software to create any size whole or you may want to add this as an option to set hole size for shaft hole in exported SVG ?

**Comment by jumpjack** [comment]
Very good point, I didn't notice it... because actually I stopped developing this code before actually creating any real cam with it, being passed to other projects.
Now the page it should work, in the exported file you will see the center marked.
I also added the ability to use the page online, without need to download anything:
https://jumpjack.github.io/CamFollowerJS/

Now I am curious to see your automata.... :-)

**Comment by makeCNCWorkshops** [comment]
wow that was super fast and super awesome...i am sorry you are no longer developing this as i have a ton of ideas and this type of software is really needed... and so far i have not seen anything like this around..would love to catch up for a phone chat with you and send you some of my research on what's out there for this type of thing ...i am Paul and you can email me paul at makecnc dot com in the first instance perhaps this may interest you [https://www.researchgate.net/publication/253136700_COMPUTER-AIDED_DESIGN_OF_MECHANICAL_AUTOMATA_ENGINEERING_EDUCATION_FOR_CHILDREN](https://www.researchgate.net/publication/253136700_COMPUTER-AIDED_DESIGN_OF_MECHANICAL_AUTOMATA_ENGINEERING_EDUCATION_FOR_CHILDREN)

**Comment by makeCNCWorkshops** [comment]
The primary person on the above project passed away and the other person Glenn has told me he will try and find the source code but not to hold my breath as he is now 71 and no longer interested in it as a project.. he said to use whatever i could gleen from the paper with his blessing to develop something similar but i do not have the coding skill it really looks like the holy grail of software for Automata makers ..when they were developing it output was for a laser in a very old format ..much has changed since then with lasers and 3d printing etc

**Comment by makeCNCWorkshops** [comment]
on top of whats in the paper they wrote he sent me this info ..if you have interest

A couple of notes that may not be in the dissertation. The code was written in Java using Swing for the graphical interface. The code that allowed components to rotate was all calculated using trig - sine and cosine functions to plot the points. And the output to the laser was generated as HPGL which is what our original laser cutter required. This could be done as DXF of G code or anything else probably as easily.

**Comment by jumpjack** [comment]
> i have not seen anything like this around..

Yes... this is why I started writing my own program: because I was not able to find anything similar around, and I searched for months!
I'll have a look at the paper.

**Comment by Tjookum** [comment]
Hi there,

I know this project is over 2 years old but I was looking online for a solution to generate some CAM profiles and I was very happy to find your program. Is there any chance you will continue working on this? Your program is by far the most intuitive I have tried and with some of the options you mentioned it would be incredibly useful for hobbyists. 

I did find a really nice MATLAB program: [cam simulator](https://www.mathworks.com/matlabcentral/fileexchange/22571-cam-simulator) with most of the same features. But being able to export to SVG and the clean GUI make your solution a lot cleaner. 

I really hope you would reconsider developing it some more. As makeCNCworkshops mentioned, there just isn't any good alternatives.

**Comment by jumpjack** [comment]
You are right, I quite abandoned the project, because it was getting quite complex to improve...
But with the arrive of Artificial Intelligence assistance in coding, I was now able to add a feature which I'm sure you will like:

https://jumpjack.github.io/CamFollowerJS/

Now **you can edit the cam profile using mouse**!  (but there's yet a little bug on "point 0" of profile).

 
...I hope this means you will share some interesting cam profiles to start also warking on my "todo list" point 1  :-)  

In the meantime I also started working on point 8; I am at good point, I am now working on load/save.

1. cams library in json format **-- waiting for users input... -**
2. visual cams library
3. visual editing of cam profile using mouse -- **DONE**
4. flat/circuar roller follower support  ![image](https://user-images.githubusercontent.com/1620953/189072510-8ed40258-0e9e-47c7-a367-2c43578d53d2.png)
5. parametric editing
6. STL direct export
7. combined motions simulator
8. rotating follower  ![image](https://github.com/user-attachments/assets/9865580f-cf5a-4b1b-987c-5fa1b5ca24a7)  **--ONGOING--**

**Comment by jumpjack** [comment]
@makeCNCWorkshops 

> The primary person on the above project passed away and the other person Glenn has told me he will try and find the source code but not to hold my breath as he is now 71 and no longer interested in it as a project.. 

Hi, how did it end? Did he/you find the source code of MachineShop software?

**Comment by moh-d-m4x** [comment]
i try to fix the "Cam center" problem with the help of AI and fix it in the fork

**Comment by jumpjack** [comment]
@moh-d-m4x  you did a great job! How can I pull your changes into my code?

**Comment by moh-d-m4x** [comment]
> [@moh-d-m4x](https://github.com/moh-d-m4x) you did a great job! How can I pull your changes into my code?

@jumpjack thank you! The "Cam center" code changes PR is here: [Now the center dot will be placed at the origin point](https://github.com/jumpjack/CamFollowerJS/pull/2)

Sorry, but I was too new to do a pull request 😅 and after that I messed with the code a lot... but now I’ve created a new branch with only that commit and requested a pull. You can review it in the link above! 👆

**Comment by jumpjack** [comment]
no, I was talking about overall changes, I like the new interface, and I think you also fixed some further bugs, so probably I could merge everything. Instead, I read in readme about new STL export feature, but can't find it.

**Comment by moh-d-m4x** [comment]
> no, I was talking about overall changes, I like the new interface, and I think you also fixed some further bugs, so probably I could merge everything. Instead, I read in readme about new STL export feature, but can't find it.

alright then i requested a pull here :[fix-exportSTL-and-deforming-3d-cam](https://github.com/jumpjack/CamFollowerJS/pull/3)
it contains the last changes and i fixed the .STL file exporter, it got removed for some reason..
you can test it here live : [Cam/Follower Mechanism Simulator](https://moh-d-m4x.github.io/CamFollowerJS/)

**Comment by jumpjack** [comment]
I can't find Issues  and Discussions on your repo, so I write it here: you may be interested in what I was able to accomplish in a couple of days thanks to Google AI studio! (of course still work in progress...)

- [Robot arm editor](https://jumpjack.github.io/automata/html/robotarm/editor/arm-editor-006.html)
- [Robot arm recorder/player](https://jumpjack.github.io/automata/html/robotarm/recorder/index.html)

Source: https://github.com/jumpjack/automata/tree/main/html/robotarm

Maybe you can find them useful for the "stickman" page.

The "recorder/player" shows an unmodifiable robot arm and allows recording handrwriting and replaying it.
The "editor" allows creating a custom robot arm structure; the idea is driving it by specifying values for angles of each joint, but as of now it loads a trajectoy created by recorder and replays it; you can try [this trajectory](https://github.com/jumpjack/automata/blob/main/html/robotarm/recorder/hello.json) and [this robot arm definition](https://github.com/jumpjack/automata/blob/main/html/robotarm/editor/robot-arm.json), manually inserting following configuration to associate angles to joints:

<img width="502" height="350" alt="Image" src="https://github.com/user-attachments/assets/50c30bbe-475b-4402-a52f-8371412d8603" />

Note that you'll get a slightly different "replay" because the arm definition is slightly different than rhe arm used for recording.

**Comment by moh-d-m4x** [comment]
very interesting, I've tried it and see the potential, i will keep the project in mind if i needed it.. thank you for letting me know about it..

and yes the AI is super helpful to getting idea\imagine to action .. you really did a good job directing it, i also did a project called [MechAnim](https://github.com/moh-d-m4x/MechAnim) with Google AI studio and it turn out as i wanted it .. its also for linkages that used in automata, but more for a circular motion..
